### ISSUE TITLE
 Type instantiation is excessively deep and possibly infinite.  

### ISSUE THREAD

**Issue by HaNdTriX** [nuxt-community/apollo-module#576]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v18.16.0
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  yarn@1.22.19
- Builder:          -
- User Config:      devtools, app, devServer, ssr, modules, css, pages, typescript, vite, runtimeConfig, routeRules, apollo
- Runtime Modules:  @nuxtjs/apollo@5.0.0-alpha.9
- Build Modules:    -
------------------------------

### Describe the bug

When running `npx nuxi typecheck` I am getting the following error after updating to `@nuxtjs/apollo@5.0.0-alpha.9`

```
nuxt.config.ts:43:12 - error TS2589: Type instantiation is excessively deep and possibly infinite.

43   modules: ['@nuxtjs/apollo'],
```

### Expected behaviour

`npx nuxi typecheck` should not raise an error

### Reproduction

Run `npx nuxi typecheck`

**Comment by FranciscoKloganB** [comment]
I also got this one on `alpha.11`

**Comment by pschaub** [comment]
I am not able to reproduce this issue.

@HaNdTriX @FranciscoKloganB Could you provide an example file `nuxt.config.ts` that results into this issue?

**Comment by HaNdTriX** [comment]
I just checked and the issue seems to be resolved in `v5.0.0-alpha.11`
